src/app/forms-builder/components/dynamic-fields/dynamic-form.component.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  AfterViewChecked,
+  AfterViewInit,
+  Component,
+  Input,
+  OnInit,
+  ViewChild,
+  ViewChildren
+} from '@angular/core';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import { fromEvent } from 'rxjs';
+import { Store } from '@ngrx/store';
+
+import { FieldConfig } from '../../../core/interfaces/field.interface';
+import { FormsBuilderContentState } from '../../../core/state/reducers';
+import { FormsBuilderService } from '../../services/forms-builder.service';
+import { DropAreaChangeIndexAction } from '../../../core/state/actions/dropAreaActions';
+import {takeUntil} from "rxjs/operators";
+
+
+@Component({
+  selector: 'app-dynamic-form',
+  templateUrl:'./dynamic-form.component.html',
+  styleUrls:[`./dynamic-form.component.scss`]
+})
+
+export class DynamicFormComponent implements OnInit,AfterViewInit,AfterViewChecked {
+
+  public form: FormGroup;
+  public changeContent$:any
+
+  @Input() public fields: FieldConfig[] = [];
+  @Input() public isStyleInput:boolean
+  @Input() public isFormActive:boolean
+  @Input() public items:any
+  @Input() public selectedIndex:any
+
+  @ViewChildren('dynamicInputs') public inputComponents:any
+  @ViewChild('dynamicForm') public dynamicForm:any
+
+
+  get value() {
+    return this.form.value;
+  }
+
+  constructor(private fb: FormBuilder,private store$:Store<FormsBuilderContentState>,private formsBuilderService:FormsBuilderService) {}
+
+  public createControl(): FormGroup {
+    const group = this.fb.group({});
+    this.fields.forEach(field => {
+      if (field.type === 'button') return;
+      const control = this.fb.control(
+        field.value
+      );
+      group.addControl(field.name!, control);
+    });
+    return group;
+  }
+
+  ngOnInit(): void {
+    this.form = this.createControl();
+  }
+  ngAfterViewInit() {
+
+  }
+  ngAfterViewChecked() {
+   this.changeContent$= fromEvent(this.dynamicForm.nativeElement.children,'change').pipe(
+     takeUntil(fromEvent(document.getElementById('logout-btn')!,'click'))
```

Comment:
use takeUntil(destroy$) here
https://stackoverflow.com/questions/42490265/rxjs-takeuntil-angular-components-ngondestroy
```suggestion
     takeUntil(this.destroy$)
```